docs/nx-cloud/enterprise/dte/github-dte.md

```diff
@@ -0,0 +1,185 @@
+# Custom Distributed Task Execution on Github Actions
+
+Using [Nx Agents](/ci/features/distribute-task-execution) is the easiest way to distribute task execution, but it your organization may not be able to use hosted Nx Agents. With an [enterprise license](https://nx.app/enterprise), you can set up distributed task execution on your own CI provider using the recipe below.
+
+## Use the GitHub Action Reusable Workflow
+
+```yaml {% fileName=".github/workflows/ci.yml" %}
+name: CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+# Needed for nx-set-shas when run on the main branch
+permissions:
+  actions: read
+  contents: read
+
+jobs:
+  main:
+    name: Nx Cloud - Main Job
+    uses: nrwl/ci/.github/workflows/nx-cloud-main.yml@v0.13.0
+    with:
+      number-of-agents: 3
+      parallel-commands: |
+        npx nx-cloud record -- nx format:check
+      parallel-commands-on-agents: |
```

Comment:
Let's use the manual GitHub config here. No one On-Prem setting up DTE is using this reusable workflow.

Add `e2e-ci`
Add `NX_CLOUD_DISTRIBUTED_EXECUTION_AGENT_COUNT: 3` env variable